src/main/java/io/github/nucleuspowered/nucleus/util/WhenTime.java

```diff
@@ -0,0 +1,199 @@
+            case SECONDS:
+                cronBuilder.withDoM(FieldExpressionFactory.always())
+                        .withDoW(FieldExpressionFactory.questionMark())
+                        .withYear(FieldExpressionFactory.always())
+                        .withMonth(FieldExpressionFactory.always())
+                        .withHour(FieldExpressionFactory.on(0))
+                        .withMinute(FieldExpressionFactory.on(0))
+                        .withSecond(FieldExpressionFactory.every(this.every));
+                break;
+            default:
+                throw new IllegalArgumentException("Time Unit is not supported by Quartz CRON.");
+        }
+        this.isSingle = false;
+        this.backingCronInstance = cronBuilder.instance();
+    }
+
+    public WhenTime(QuartzTimeUnit timeUnit, int every, boolean isSingle) throws IllegalArgumentException {
+        this(timeUnit, every);
+        this.isSingle = isSingle;
+    }
+
+    /**
+     * @return
+     *  If the user specified to only run this once
+     */
+    public boolean isSingle() {
+        return this.isSingle;
+    }
+
+    /**
+     * Grab the backing cron instance for this WhenTime
+     *
+     * @return
+     *  The backing cron instance
+     */
+    public Cron getBackingCronInstance() {
+        return this.backingCronInstance;
+    }
+
+    /**
+     * Converts the WhenTime to a String
+     *
+     * @return
+     *  The WhenTime as a String
+     */
+    public String asString() {
+        if (!this.isSingle) {
+            if (this.every == 1) {
+                return String.format("every %s", this.timeUnit.name().toLowerCase());
+            } else {
+                return String.format("every %d %s", this.every, this.timeUnit.name().toLowerCase());
+            }
+        } else {
+            if (this.every == 1) {
+                return String.format("in %s", this.timeUnit.name().toLowerCase());
+            } else {
+                return String.format("in %d %s", this.every, this.timeUnit.name().toLowerCase());
+            }
+        }
+    }
+
+    /**
+     * Parses a WhenTime from a String
+     *
+     * @param string
+     *  The String to parse
+     * @return
+     *  The WhenTime represented by the String
+     * @throws IllegalArgumentException
+     *  Throws an IllegalArgumentException if we could not parse the string
+     * @throws NullPointerException
+     *  Throws a NullPointerException if the String is null
+     */
+    public static WhenTime fromString(String string) throws IllegalArgumentException, NullPointerException {
+        Preconditions.checkNotNull(string);
+        if (!string.contains(" ")) {
+            throw new IllegalArgumentException("Expression doesn't have the split character.");
+        }
+        String[] split = string.split(" ");
+        if (split.length > 3 || split.length < 2) {
+            throw new IllegalArgumentException("Expression doesn't has to many directives.");
```

Comment:
Yea looking at it now, I'm not sure what I was thinking when I wrote that exception message.